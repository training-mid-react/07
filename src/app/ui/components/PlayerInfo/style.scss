@use '../../styles/mixins' as mix;

.player-info {
    gap: 10px;
    align-items: start;

    &_item {
      gap: 5px;
      @include mix.flex();
    }
  }