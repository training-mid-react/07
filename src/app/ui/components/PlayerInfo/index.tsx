import React from 'react';
import Title from '@components/Title';
import './style.scss';

const players = [
  { name: 'Player 1', cellClass: 'cell-1' },
  { name: 'Player 2', cellClass: 'cell-2' },
];

const PlayerInfo: React.FC = () => {
  return (
    <div className="player-info" data-testid="player-info">
      {players.map((player) => (
        <div key={player.name} className="player-info_item" data-testid={`player-info_item-${player.cellClass}`}>
          <Title as="h4" data-testid={`player-info_item-${player.cellClass}`}>{player.name}</Title>
          <div className={`cell ${player.cellClass}`}data-testid={`player-cell-${player.name}`}></div>
        </div>
      ))}
    </div>
  );
};

export default PlayerInfo;
