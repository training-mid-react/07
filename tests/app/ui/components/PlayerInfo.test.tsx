import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import PlayerInfo from '../../../../src/app/ui/components/PlayerInfo';

describe('Componente PlayerInfo', () => {
  it('renderiza correctamente', () => {
    const { asFragment } = render(<PlayerInfo />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renderiza el contenedor principal con el data-testid correcto', () => {
    const { getByTestId } = render(<PlayerInfo />);
    const playerInfoElement = getByTestId('player-info');
    expect(playerInfoElement).toBeInTheDocument();
  });

  it('renderiza el Ã­tem de cada jugador con el data-testid correcto', () => {
    const { getByTestId } = render(<PlayerInfo />);
    const player1Item = getByTestId('player-info_item-cell-1');
    const player2Item = getByTestId('player-info_item-cell-2');
    expect(player1Item).toBeInTheDocument();
    expect(player2Item).toBeInTheDocument();
  });

  it('renderiza los nombres de los jugadores correctamente', () => {
    const { getByText } = render(<PlayerInfo />);
    expect(getByText('Player 1')).toBeInTheDocument();
    expect(getByText('Player 2')).toBeInTheDocument();
  });

  it('asigna las clases correctas a las celdas de los jugadores', () => {
    const { container } = render(<PlayerInfo />);
    const player1Cell = container.querySelector('.cell-1');
    const player2Cell = container.querySelector('.cell-2');
    expect(player1Cell).toBeInTheDocument();
    expect(player2Cell).toBeInTheDocument();
  });

  it('renderiza el componente Title correctamente', () => {
    const { getByTestId } = render(<PlayerInfo />);
    const titlePlayer1 = getByTestId('player-info_item-cell-1').querySelector('h4');
    const titlePlayer2 = getByTestId('player-info_item-cell-2').querySelector('h4');
    expect(titlePlayer1).toBeInTheDocument();
    expect(titlePlayer2).toBeInTheDocument();
  });

  it('renderiza las celdas de los jugadores con el data-testid correcto', () => {
    const { getByTestId } = render(<PlayerInfo />);
    const player1Cell = getByTestId('player-cell-Player 1');
    const player2Cell = getByTestId('player-cell-Player 2');
    expect(player1Cell).toBeInTheDocument();
    expect(player2Cell).toBeInTheDocument();
  });
});
