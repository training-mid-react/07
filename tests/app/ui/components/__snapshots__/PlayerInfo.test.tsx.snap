// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente PlayerInfo > renderiza correctamente 1`] = `
<DocumentFragment>
  <div
    class="player-info"
    data-testid="player-info"
  >
    <div
      class="player-info_item"
      data-testid="player-info_item-cell-1"
    >
      <h4
        class="title"
        data-testid="title-component"
      >
        Player 1
      </h4>
      <div
        class="cell cell-1"
        data-testid="player-cell-Player 1"
      />
    </div>
    <div
      class="player-info_item"
      data-testid="player-info_item-cell-2"
    >
      <h4
        class="title"
        data-testid="title-component"
      >
        Player 2
      </h4>
      <div
        class="cell cell-2"
        data-testid="player-cell-Player 2"
      />
    </div>
  </div>
</DocumentFragment>
`;
